### ISSUE TITLE
Proposal: add API should not infer car files

### ISSUE THREAD

**Issue by Gozala** [nftstorage/ipfs-cluster#12]
At the moment when you call `cluster.add(file)` if file is `application/car` or if it's name ends with `.car` it will be imported as car file.

https://github.com/nftstorage/ipfs-cluster/blob/0458e04c0b0331fab4f08f0c092b774c71c0ead2/index.js#L43-L45

This proved error prone as per https://github.com/ipfs-shipyard/nft.storage/issues/434, because user provided files may have names or mime types that would affect behavior and invalidate held invariants like:

https://github.com/ipfs-shipyard/nft.storage/blob/60499653c4ebb4ad4590527243a09bc8bb73c540/packages/api/src/cluster.js#L59-L62
```js
export const importAsset = async (file, options = {}) => {
  const result = await client.addDirectory([file], options)
  if (result.length !== 2) {
    throw new Error(
      `Expected response with two entries, but got ${result.length} instead`
    )
  }
  const [, dir] = result
  return dir.cid
}
```

We could clear mime type if we wish to add files as files as opposed to dag archives, but changing a file name seems a bit more extreme.

Additionally some of the options that `client.add` takes make no sense in the contex of the dag import 

https://github.com/nftstorage/ipfs-cluster/blob/0458e04c0b0331fab4f08f0c092b774c71c0ead2/index.d.ts#L83-L110

And the whole thing is even more confusing when you do `client.addDirectory([new File([], 'love.car')` which will not create directory with a file but would import dag and not sure if it does something with directory

https://github.com/nftstorage/ipfs-cluster/blob/0458e04c0b0331fab4f08f0c092b774c71c0ead2/index.js#L67-L72

**Comment by vasco-santos** [comment]
I also agree that the add API should not make such decisions under the hood, specially when conflicts might occur with the decision factors.

@Gozala were you expecting a new API for adding CARs, or an explicit option provided by the user on add?

We still need to figure out what the ideal solution would be for directories, specially as they could also have both CAR and non CAR files
